movie_maker and notebooks

DriftKing1998 opened the pull request:
- added movie_maker to create gifs
- added notebooks as tutorial
- gifs were not able to be added to notebook

DriftKing1998 committed:
movie_maker and notebooks

- added movie_maker to create gifs
- added notebooks as tutorial
- gifs were not able to be added to notebook

DriftKing1998 committed:
authors

akshey-kumar commented on demos/cognitive_graphs.ipynb:
In the seciton 'Interactive plots (using pyvis)',
figures/test.html
Plot has been saved under: /Users/michaelhofer/Documents/GitHub/UniProject/figures/test.html

I cannot find test.html in the figures directory. Perhaps you saved it somewhere outside the NC-MCM repo and thus it did not get pushed? Also, it would be preferable to use the relative path with respect to the NC-MCM directory full, rather than the full path that would vary based on each user.

akshey-kumar commented on figures/Interactive_graph.html:
![image](https://github.com/user-attachments/assets/2427abba-eace-4211-8ba4-64fce8fdaba1)

I like the idea of having an interactive plot. However, I am not sure if all the sliders work as expected. Please carefully check before pushing. Also, what are these other parameters doing there (see image)?

akshey-kumar commented on ncmcm/visualisers/tests/comparison.png:
How does the clasifier achieve an accuracy of 1.0 when the data does not seem to have any structure with respect to the behaviour? How are there no incorrect predictions? That seems counterintuitve.

 Kindly explain the use of this example in our meeting.

DriftKing1998 committed:
Added changes

- comparison plot shows in-sample & cross-validation accuracy

- I am using relative paths

- fixed the menu for interactive plots, by removing all setting and just keeping the selection mode

- added options for physics (0 to 2) aswell as the possibility to use own json-file for physics

DriftKing1998 committed:
News

DriftKing1998 committed:
Merge branch 'main' into michaels-branch

DriftKing1998 committed:
Fixed tests for changes

DriftKing1998 committed:
fixed tests

DriftKing1998 committed:
refactoring

- renamed methods
- added parameter to choose test mode
- redid cognitive graph notebook

DriftKing1998 committed:
Better Plot
